Add Flyout guidelines and docs structure (#1057)
hashicorp/design-system: website/docs/components/flyout/partials/guidelines/overview.md

A Flyout is used to show and hide additional information and content that is related to the main page without navigating or routing to a new page.

## Usage

### When to use

When displaying additional information, context, or details about an object or element present on the main page.

### When not to use

- When requesting the user for information or feedback, use a [Modal](/components/modal).

> - When displaying overly complex information, consider moving the content to it’s own page.

Reviewer: ```suggestion
- When displaying overly complex information, consider moving the content to its own page.
```
